### ISSUE TITLE
Prop type when hyphenated and attribute type regression in 1.3.9

### ISSUE THREAD

**Issue by markglattback** [vuejs/language-tools#2587]
In the example below, `PInputText` accepts a `modelValue` prop which is now being highlighted as a type error **when passed in hyphenated form**. The `type` attribute (not defined as a prop inside of the component but is consumed internally), is also getting flagged as an error when it wasn't previously.

![image](https://user-images.githubusercontent.com/10074497/230028123-7cc0e6d5-1e3f-455d-b2af-6644876cea70.png)

The error message for both is:

```ts
Type 'string' is not assignable to type 'Partial<Record<any, (_: any) => any>> & Partial<Record<any, (_: any) => any>>'.
  Type 'string' is not assignable to type 'Partial<Record<any, (_: any) => any>>'
```

This looks like a regression in 1.3.9 as it' was working previously in 1.3.8.

**Comment by rchl** [comment]
It might not be the root issue but conceptually passing a type property to a component that doesn't accept such prop sounds very wrong. The user of the component should not assume any specific structure within the component. In this case it's being assumed that the root element of the component is an input element. The component should instead accept such prop explicitly.

**Comment by markglattback** [comment]
Yes, in this particular instance `type` could, and maybe _should_ be a prop, but this component is used as part of an internal library where we have documented the ability to pass through attributes to the underlying `input` (which doesn't happen to be the root element, but I'm not sure if that's relevant?).

Further testing has revealed that the above errors are only thrown when props are defined using TS (`defineProps<Props>()`) inside of PInputText. Using a JS object (`defineProps({ ...props })`) doesn't cause errors to be highlighted:

![image](https://user-images.githubusercontent.com/10074497/230060236-464f190d-e982-49fa-94cf-1f58d2857b70.png)
